internal/venice-common/src/main/java/com/linkedin/venice/hadoop/schema/HDFSSchemaSource.java

```diff
@@ -121,6 +155,22 @@ void saveSchemaResponseToDisk(MultiSchemaResponse.Schema[] schemas, boolean isRm
     }
   }
 
+  void saveKeySchemaToDisk(SchemaResponse schema) throws IOException, IllegalStateException {
+    Preconditions.checkState(includeKeySchema, "Cannot be invoked with invalid key schema directory");
```

Comment:
Should we also check if the `SchemaResponse` has any errors before processing it?